### PR TITLE
 * Switch to new logging style

### PR DESCRIPTION
**Pull request by ryanbrandenburg**

### CODE REVIEW

File: src/Microsoft.AspNet.Routing/Logging/RouterMiddlewareLoggerExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+namespace Microsoft.AspNet.Routing.Logging
+{
+    internal static class RouterMiddlewareLoggerExtensions
+    {
+        private static readonly Action<ILogger, Exception> _requestDidntMatchRoutes;
+
+        static RouterMiddlewareLoggerExtensions()
+        {
+            _requestDidntMatchRoutes = LoggerMessage.Define(LogLevel.Debug, 1,
```
Comment (pranavkm) [inline src/Microsoft.AspNet.Routing/Logging/RouterMiddlewareLoggerExtensions.cs]: Params go on individual lines.

File: src/Microsoft.AspNet.Routing/Logging/RouterMiddlewareLoggerExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNet.Routing.Logging
+{
+    internal static class RouterMiddlewareLoggerExtensions
+    {
+        private static readonly Action<ILogger, Exception> _requestDidntMatchRoutes;
```
Comment (pranavkm) [inline src/Microsoft.AspNet.Routing/Logging/RouterMiddlewareLoggerExtensions.cs]: `_requestDidNotMatchRoutes` (the lack of the apostrophe makes Didnt look weird)

**Comment by pranavkm** [comment]
:shipit:

File: src/Microsoft.AspNet.Routing/Logging/LogFormatter.cs [hunk 3]
```diff
@@ -1,33 +0,0 @@
...
-using System;
-
-namespace Microsoft.AspNet.Routing.Logging.Internal
-{
-    public static class LogFormatter
-    {
-        /// <summary>
-        /// A formatter for use with <see cref="Microsoft.Extensions.Logging.ILogger.Log"/>.
-        /// </summary>
-        public static string Formatter(object o, Exception e)
```
Comment (rynowak) [inline src/Microsoft.AspNet.Routing/Logging/LogFormatter.cs]: #JustBenBroThings

File: src/Microsoft.AspNet.Routing/RouteBase.cs [hunk 4]
```diff
@@ -105,13 +106,9 @@ public virtual Task RouteAsync(RouteContext context)
             {
                 return TaskCache.CompletedTask;
             }
+            _logger.MatchedRouteName(Name, ParsedTemplate.TemplateText);
```
Comment (rynowak) [inline src/Microsoft.AspNet.Routing/RouteBase.cs]: This isn't right, it's `MatchedRoute`. We're just logging the name as part of teh dataz

**Pull request merged**